pystac_client/collection_client.py

```diff
@@ -174,3 +173,31 @@ def get_item(self, id: str, recursive: bool = False) -> Optional["Item_Type"]:
             call_modifier(self.modifier, item)
 
         return item
+
+    def get_queryables(self) -> Dict[str, Any]:
+        """Return all queryables.
+
+        Output is a dictionary that can be used in ``jsonshema.validate``
+
+        Return:
+            Dict[str, Any]: Dictionary containing queryable fields
+        """
+        root = self.get_root()
+        assert root
+        stac_io = root._stac_io
+        assert stac_io
+        assert isinstance(stac_io, StacApiIO)
+
+        stac_io.assert_conforms_to(ConformanceClasses.FILTER)
+
+        self_href = self.get_self_href()
+        if self_href is None:
+            raise ValueError("cannot build a queryable href without a self href")
+
+        url = f"{self_href.rstrip('/')}/queryables"
+
+        result = stac_io.read_json(url)
+        if "properties" not in result:
+            raise APIError("Invalid response from /queryables")
```

Comment:
Yeah, this feels pretty un-DRY. You mentioned a mixin was too wild from the typing side of things...is there anything we can do with free functions to share behavior between the two implementations?

Reply:
yeah that's a good idea. I really did like the Mixin. It felt kind of like a good match for the "conformsTo" extension system. Maybe I should give that one more try. I'll dry it up either way.